fbw-a32nx/src/systems/instruments/src/EWD/NewPseudoFWC.ts

```diff
@@ -0,0 +1,2253 @@
+    }
+
+    masterWarning(toggle: number) {
+        SimVar.SetSimVarValue('L:A32NX_MASTER_WARNING', 'Bool', toggle);
+        SimVar.SetSimVarValue('L:Generic_Master_Warning_Active', 'Bool', toggle);
+    }
+
+    masterCaution(toggle: number) {
+        SimVar.SetSimVarValue('L:A32NX_MASTER_CAUTION', 'Bool', toggle);
+        SimVar.SetSimVarValue('L:Generic_Master_Caution_Active', 'Bool', toggle);
+    }
+
+    mapOrder(array, order): [] {
+        array.sort((a, b) => {
+            if (order.indexOf(a) > order.indexOf(b)) {
+                return 1;
+            }
+            return -1;
+        });
+        return array;
+    }
+
+    private adirsMessage1(adirs, engineRunning) {
+        let rowChoice = 0;
+
+        switch (true) {
+        case (Math.ceil(adirs / 60) >= 7 && !engineRunning):
+            rowChoice = 0;
+            break;
+        case (Math.ceil(adirs / 60) >= 7 && engineRunning):
+            rowChoice = 1;
+            break;
+        case (Math.ceil(adirs / 60) === 6 && !engineRunning):
+            rowChoice = 2;
+            break;
+        case (Math.ceil(adirs / 60) === 6 && engineRunning):
+            rowChoice = 3;
+            break;
+        case (Math.ceil(adirs / 60) === 5 && !engineRunning):
+            rowChoice = 4;
+            break;
+        case (Math.ceil(adirs / 60) === 5 && engineRunning):
+            rowChoice = 5;
+            break;
+        case (Math.ceil(adirs / 60) === 4 && !engineRunning):
+            rowChoice = 6;
+            break;
+        case (Math.ceil(adirs / 60) === 4 && engineRunning):
+            rowChoice = 7;
+            break;
+        default:
+            break;
+        }
+
+        return rowChoice;
+    }
+
+    private adirsMessage2(adirs, engineRunning) {
+        let rowChoice = 0;
+
+        switch (true) {
+        case (Math.ceil(adirs / 60) === 3 && !engineRunning):
+            rowChoice = 0;
+            break;
+        case (Math.ceil(adirs / 60) === 3 && engineRunning):
+            rowChoice = 1;
+            break;
+        case (Math.ceil(adirs / 60) === 2 && !engineRunning):
+            rowChoice = 2;
+            break;
+        case (Math.ceil(adirs / 60) === 2 && engineRunning):
+            rowChoice = 3;
+            break;
+        case (Math.ceil(adirs / 60) === 1 && !engineRunning):
+            rowChoice = 4;
+            break;
+        case (Math.ceil(adirs / 60) === 1 && engineRunning):
+            rowChoice = 5;
+            break;
+        default:
+            break;
+        }
+
+        return rowChoice;
+    }
+
+    onUpdate(deltaTime) {
+        // Inputs update
+
+        const flightPhaseInhibitOverride = SimVar.GetSimVarValue('L:A32NX_FWC_INHIBOVRD', 'bool');
+
+        this.fwc1Normal.set(SimVar.GetSimVarValue('L:A32NX_FWS_FWC_1_NORMAL', 'bool'));
+        this.fwc2Normal.set(SimVar.GetSimVarValue('L:A32NX_FWS_FWC_2_NORMAL', 'bool'));
+
+        /*
+        if (!this.fwc1Normal.get() && !this.fwc2Normal.get()) {
```

Comment:
Commented code. Remove if not needed

Reply:
This PR was originally targeted at experimental, and experimental implements the warning computers in Rust, which also adds the "FWC normal" simvars.

I copied over the functionality to allow the system to show FWC faults, but since master does not have this system, it caused false positives. I thought it might be best to leave the code in so that when we add it to experimental it's a simple task to re-enable them.

You're right though, it might be better to add this functionality to the Rust FWC PR instead, once this one gets merged.

Reply:
Oh I see, I'm too used to experimental here. In that case do as you wish 👍